MC partilce information

stefanopolitano opened the pull request:
This PR contains:

- pdg information for several MC particles

fgrosa commented on flarefly/utils.py:
```diff
@@ -42,3 +42,144 @@ def __init__(self, text, level):
+        elif self._pdg_ == 3334:
+            pdg_dict = {'name': 'Omega-', 'mass': 1.67245}
+        else:
+            pdg_dict = {'name': 'unknown', 'mass': 0.0}
```
This line does not work anyway because `FATAL` exits from the execution
```suggestion
```

fgrosa commented on flarefly/utils.py:
```diff
@@ -42,3 +42,144 @@ def __init__(self, text, level):
+        Parameters
+        -------------------------------------------------
+        unit: str
+            Unit of the mass, possible values ['GeV', 'MeV']
```
Missing documentation for the "Returns"

fgrosa commented on flarefly/utils.py:
```diff
@@ -42,3 +42,144 @@ def __init__(self, text, level):
+        ------------------------------------------------
+        pdg:int
+            PDG code of the particle
+        """
```
Missing documentation for the "Returns"

fgrosa commented on flarefly/utils.py:
```diff
@@ -42,3 +42,144 @@ def __init__(self, text, level):
+    def get_name(self):
+        """
+        Get particle name
+        """
```
Missing documentation for the returns

fgrosa reviewed (changes requested):
Hi @stefanopolitano I am not sure that this class is needed, it would have been enough to have a dictionary that relates name and mass (or pdg code and mass) that can be used everywhere in the code (it's not optimal in my opinion to build an object everytime we need a mass). Anyway, I would also add a functionality here
https://github.com/flarefly/flarefly/blob/fccf967077c70cb1644f34799b91dc00db662876/flarefly/fitter.py#L100
that allows the user to pass either the pdg code or the particle name and, provided this information, sets the correct mass as starting point for the fit

stefanopolitano commented:
Hi @fgrosa! Thanks for the review, I changed the class into a dictionary and removed the functions of the class. I also prepared a function that provides the particle mass if the id or the name of the particle is provided: I am not sure whether it should be better to have this function in `utils.py` or in `fitter.py`. Anyhow, I don't have a strong opinion on this

fgrosa commented:
> Hi @fgrosa! Thanks for the review, I changed the class into a dictionary and removed the functions of the class. I also prepared a function that provides the particle mass if the id or the name of the particle is provided: I am not sure whether it should be better to have this function in `utils.py` or in `fitter.py`. Anyhow, I don't have a strong opinion on this

Hi @stefanopolitano, thanks! I would put this function in the utils, since it might be useful for other modules in the future.

fgrosa commented on flarefly/utils.py:
```diff
@@ -44,3 +44,70 @@ def __init__(self, text, level):
+            '3312': {'name': 'Xi-', 'mass': 1.32132},
+            '3334': {'name': 'Omega-', 'mass': 1.67245}}
+
+def get_particle_mass(pdg_id=None, pdg_name=None): # pylint: disable=inconsistent-return-statements
```
For the time being we can merge it like this, but I would then change it in the future because in this was both arguments are needed (while we can just put kwargs)

fgrosa commented on flarefly/utils.py:
```diff
@@ -44,3 +44,70 @@ def __init__(self, text, level):
         else:
             print(text)
+
+pdg_dict = {'22': {'name': 'photon', 'mass': 0.0},
```
the key does not need to be a string (but we can live with that for the moment`)
